cranelift/codegen/src/isa/x64/inst.isle

```diff
@@ -3422,32 +3448,35 @@
 ;; Ensure that we put the `x` argument into a register for single-register
 ;; gpr-typed arguments, as we rely on this for the legalization of heap_addr and
 ;; loading easily computed constants (like 0) from memory is too expensive.
-(rule (select_icmp (IcmpCondResult.Condition producer cc) x @ (value_type (is_gpr_type (is_single_register_type ty))) y)
+(rule (select_icmp (IcmpCondResult.Condition producer cc) x @ (value_type (is_single_register_gpr_type ty)) y)
       (with_flags producer (cmove ty cc (put_in_gpr x) y)))
 
 ;; Otherwise, fall back on the behavior of `cmove_from_values`.
-(rule (select_icmp (IcmpCondResult.Condition producer cc) x @ (value_type ty) y)
+(rule (select_icmp (IcmpCondResult.Condition producer cc) x @ (value_type (is_multi_register_gpr_type ty)) y)
+      (with_flags producer (cmove_from_values ty cc x y)))
+
+(rule (select_icmp (IcmpCondResult.Condition producer cc) x @ (value_type (is_xmm_type ty)) y)
```

Comment:
Hmmm, is it better to duplicate the rule like this, or to put the single-gpr case at a higher priority?

Reply:
In the future, I'd love to use an or-pattern here. Given the current compilation for rules at different priorities, I'm tempted to say that duplication for rules with a small RHS is preferable. I have an idea that I'd like to try out for using the overlap information to allow earlier match reuse in the trie, and in that case using different priorities wouldn't be nearly as costly.

Reply:
I guess I don't feel strongly about duplication for rules with a small RHS. I'm just more concerned about maintainability of the ISLE source than I am about the performance of the ISLE-generated code right now. We know we'll fix the latter by improving the ISLE compiler, but the former is harder to recover.

Reply:
OK, I've switched to resolving this with priorities.